lib/ClusterShell/Worker/Rsh.py

```diff
@@ -59,8 +60,25 @@ def _build_cmd(self):
         cmd_l.append("%s" % self.key)  # key is the node
         cmd_l.append("%s" % self.command)
 
+        # rsh does not properly return exit status
+        # force the exit status to be printed out
+        cmd_l.append("; echo XXRETCODE: $?")
+
         return (cmd_l, None)
 
+    def _on_nodeset_msgline(self, nodes, msg, sname):
+        """local wrapper over _on_node_msgline that can also handle nodeset"""
+        match = re.search("^XXRETCODE: (\d+)$", msg.decode("utf-8"))
```

Comment:
utf8 handling is a pain :) @thiell thoughts on that part?

Reply:
I would rather use `decode()` without argument, like in the [Pdsh worker](https://github.com/cea-hpc/clustershell/blob/master/lib/ClusterShell/Worker/Pdsh.py#L139). Default encoding is `'utf-8'` anyway. Note that by default the `errors` argument of `decode()` is set to `'strict'` which may raise a UnicodeError exception, but that is fine with me here as the output text encoding after XXRETCODE is expected.

Reply:
Done.  Thanks!